protocol/bin/js-code-generators.js

```diff
@@ -17,17 +17,26 @@ const channelMapPrelude = `\nfunction _channelMapRpcHelper(channelConfig: Channe
 }
 `
 
-function rpcChannelMap (methodName, name, callbackType, innerParamType, responseType) {
-  return `\nexport function ${name}RpcChannelMap (configKeys: Array<string>, request: $Exact<${['requestCommon', callbackType, innerParamType].filter(t => t).join(' & ')}>): EngineChannel {
+function rpcChannelMap(methodName, name, callbackType, innerParamType, responseType) {
+  return `\nexport function ${name}RpcChannelMap (configKeys: Array<string>, request: $Exact<${[
```

Comment:
:sheep: Good idea to clean this up. Might be worth taking it one step further and breaking this expression out into a variable assignment above the template